scripts/join_ceph_cluster.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+(( counter=0 ))
+
+host=$(hostname)
+
+truncate --size=0 ~/.ssh/known_hosts  2>&1
+
+/srv/cray/scripts/common/pre-load-images.sh
+
+function gather_ceph_conf () {
+  FSID=$(ssh $node 'ceph -s --format=json-pretty|jq -r .fsid')
+  WAS_MON=$(ssh $node ceph node ls|jq -r --arg h $host 'any(.mon|keys; .[] == $h)')
+  WAS_OSD=$(ssh $node ceph node ls|jq -r --arg h $host 'any(.osd|keys; .[] == $h)')
+  if $WAS_OSD
+  then
+    OSDS+=($(ssh $node "ceph osd ls-tree $host"))
+  fi
+  CONF=$(ssh $node ceph config generate-minimal-conf)
+  echo "fsid $FSID"
+  echo "OSDS ${OSDS[@]}"
+  echo "WAS_MON $WAS_MON"
+  echo "WAS_OSD $WAS_OSD"
+
+}
+
+function apply_ceph_conf () {
+  if [[ -n $WAS_OSD ]]
+  then
+    if [[ ! -d /var/lib/ceph/$FSID ]]
+    then
+      mkdir /var/lib/ceph/$FSID
+    fi
+    for osd in ${OSDS[@]}
+    do
+      if [[ ! -d /var/lib/ceph/$FSID/osd.$osd ]]
+      then
+       mkdir /var/lib/ceph/$FSID/osd.$osd
+      fi
+      echo "$CONF" > /var/lib/ceph/$FSID/osd.$osd/config
```

Comment:
Should we fail if this write doesn't succeed (for example, if any of the mkdirs fail)?

Reply:
fixed